### DISCUSSION TITLE
Listing of which clojure.core functions are implemented

### DISCUSSION THREAD

**Discussion by stevemolitor** [squint-cljs/cherry#108] [Q&A] [answered]
Is there a listing or source file I can check to see which `clojure.core` functions are implemented? Apparently `clojure.walk` is not:

```clojure
(ns ai-poll
  (:require ["openai" :as OpenAI]
            [clojure.walk :as w]))

(def openai (OpenAI/OpenAI.))

(defn ^:async ask-ai [question temp]
  (let [payload (clj->js {:messages [{:role "user", :content question}] :model "gpt-4", :temperature temp})
        response (js/await (.chat.completions.create openai payload))
        {{answer :content} :message} (first (:choices (w/keywordize-keys (js->clj response))))]
    answer))
```

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'clojure.walk' imported from /Users/stephenmolitor/repos/ai-poll-cherry/ai_poll.mjs
    at new NodeError (node:internal/errors:405:5)
```

Context: I want to convert the response here from string map keys to keyword keys.

**Comment by stevemolitor** [comment]
I should add both cherry and squint are super fun btw! It's so great to be able to just leverage the npm ecosystem but use s expressions! Love it.

**Comment by borkdude** [comment]
We can add clojure.walk, it's currently not packaged. clojure.string is, but we can do clojure.walk in a similar fashion. Would you like to try a PR or would you like me to do it?

**Reply by stevemolitor** [reply]
Dang what a fast response thanks!

Happy to do a PR if you point me in the right direction. Clojure newbie here. Getting ready to go to work, it may be a few days. But happy to do it.

**Comment by borkdude** [comment, marked as answer]
Just added `clojure.string` (it turned out it wasn't there yet) and `clojure.walk`

**Reply by stevemolitor** [reply]
Works great, thanks a bunch!
